### PR TITLE
Fix the bml trigger on dev-env PRs

### PR DESCRIPTION
**Pull request by adilGhaffarDev**
fixes the bml trigger on dev-env

### CODE REVIEW

**Comment by kashifest** [comment]
/approve
I don’t know if the test is properly checking this peace of code.

**Comment by kashifest** [comment]
/approve cancel
It seems there is some offline discussion ongoing on the usage of this.

File: jenkins/scripts/bare_metal_lab/deploy-lab.yaml [hunk 1]
```diff
@@ -141,7 +141,7 @@
         version: "main"
       tags: git
 
-    - name: Merge branch for PR
+    - name: Configure branch for PR
```
Comment (lentzi90) [inline jenkins/scripts/bare_metal_lab/deploy-lab.yaml:144]: nit
```suggestion
    - name: Configure remote for PR
```

**Review by lentzi90 — COMMENTED** [review]
/lgtm

**Review by lentzi90 — COMMENTED** [review]
/lgtm

**Review by Sunnatillo — COMMENTED** [review]
/lgtm

**Comment by adilGhaffarDev** [comment]
> So, git rev-parse did not work for this case?

yes, test/ was required for this too. We decided to drop "test/" since that scenario only happens when we manually change branch name here https://github.com/metal3-io/project-infra/blob/65605fca5a6f6b11acea256ffcc71fe2c7fb48b9/jenkins/scripts/bml_integration_test.sh#L32 for testing, and in normal cases we will not see the branch name, we'll only see commit hash.

**Review by furkatgofurov7 — APPROVED** [review]

**Pull request merged**